kernel/mellanox_ethernet/src/lib.rs

```diff
@@ -0,0 +1,138 @@
+//! This crate defines the layout of memory objects that make up the software interface between the Mellanox hardware and the driver,
+//! as well as functions to access different fields of these objects.
+//!
+//! The Mellanox ethernet card is referred to as both the NIC (Network Interface Card) and the HCA (Host Channel Adapter).
+//!
+//! All information is taken from the Mellanox Adapters Programmer’s Reference Manual (PRM) [Rev 0.54], unless otherwise specified. 
```

Comment:
can you provide a link to the PRM? Or is it under NDA

Reply:
I can provide a link to an older version, the one I have I got directly from Nvidia.

Reply:
that's good. Explain that link in the comments and then also maybe provide a link on how to obtain the one under NDA